draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1440,9 +1445,8 @@ user agent MUST process the cookie as follows:
 15. If the cookie's `same-site-flag` is not "None":
 
     1.  If the cookie was received from a "non-HTTP" API, and the API was called
-        from a context whose "site for cookies" is not an exact match for
-        request-uri's host's registrable domain, then abort these steps and
-        ignore the newly created cookie entirely.
+        from a context whose "site for cookies" is not same-site, then abort 
```

Comment:
same-site with what? (I don't actually know how best to phrase this...)

I think we need to resolve the broader question of how to define SameSite behavior for non-HTTP APIs.

Reply:
> I think we need to resolve the broader question of how to define SameSite behavior for non-HTTP APIs.

Yeahhhhhhhh.

Until then I think what we're trying to capture is something like:

> "the origin of the calling API's "site for cookies" is same-site with the request-uri's origin?

We could say something like that, or possibly:

> If the cookie was received from a "non-HTTP" API, and the API was not called from a first-party context, then abort these steps and ignore the newly created cookie entirely.

"first-party context" *is* used in the RFC elsewhere, but it's not defined. It is in https://tools.ietf.org/html/draft-west-first-party-cookies-04#section-2.1... 🤔 

It seems like a lot of machinery to bring in just to use it here, unless we want to take that route for the further non-HTTP API route.

Reply:
I think we can do this more simply by passing in the "site for cookies" of the context (which I take to mean a document...) and the "site for cookies" of the "top-level origin" into the same-site algorithm.